### PR TITLE
BrowserToolbarView cleanup

### PR DESCRIPTION
**Pull request by NotWoods**
Again splitting up code into multiple classes

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by ekager** [comment]
Same as the other PR - can you resolve conflicts and make sure CI is passing and then assign me for review? Thank you!

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt [hunk 1]
```diff
@@ -378,15 +376,9 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
...
-                    positiveButtonBackgroundColor = ThemeManager.resolveAttribute(
-                        R.attr.accent,
-                        context
-                    ),
-                    positiveButtonTextColor = ThemeManager.resolveAttribute(
-                        R.attr.contrastText,
-                        context
-                    ),
-                    positiveButtonRadius = (resources.getDimensionPixelSize(R.dimen.tab_corner_radius)).toFloat()
+                    positiveButtonBackgroundColor = context.getColorFromAttr(R.attr.accent),
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:379]: So I seem to remember that we are doing this differently in some places for a reason... They're not equivalent depending on what the receiver is expecting. Ie look at these right next to each other: https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt#L48-L51

If we want to change these, please test thoroughly.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarPopupWindow.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package org.mozilla.fenix.components.toolbar
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarPopupWindow.kt:1]: nit - license

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt (hunk 1 above)
Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:379]: I've tested this helper and I wrote it to move `ThemeManager.resolveAttribute` into AC. The implementation is exactly the same so it should work fine.

Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:379]: Wait scratch that, I'm confusing `getColor` with `resolveAttribute`. I'll fix it here.

**Comment by ekager** [comment]
Looks like a lot of the UI tests are failing on variants of:
``` 
java.lang.NullPointerException: Attempt to invoke virtual method 'void com.google.android.material.appbar.AppBarLayout.addOnOffsetChangedListener(com.google.android.material.appbar.AppBarLayout$OnOffsetChangedListener)' on a null object reference
at org.mozilla.fenix.components.toolbar.BrowserToolbarView.<init>(BrowserToolbarView.kt:84)
at org.mozilla.fenix.browser.BaseBrowserFragment.initializeUI(BaseBrowserFragment.kt:310)
at org.mozilla.fenix.browser.BrowserFragment.initializeUI(BrowserFragment.kt:70)
at org.mozilla.fenix.browser.BaseBrowserFragment.onViewCreated(BaseBrowserFragment.kt:172)
at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:332)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1187)
at androidx.fragment.app.FragmentManager.addAddedFragments(FragmentManager.java:2224)
at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1997)
at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1953)
at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
at android.os.Handler.handleCallback(Handler.java:873)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:193)
at android.app.ActivityThread.main(ActivityThread.java:6669)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
 ```

**Comment by NotWoods** [comment]
Closing as ToolbarPopupWindow handles the bulk of this now

**Pull request closed without merging**